intermediate_source/inductor_debug_cpu.py

```diff
@@ -0,0 +1,569 @@
+# Inductor CPU backend has the support to report the time of the fusion kernels to the profiler with the ``enable_kernel_profile`` configuration option:
+
+from torch._inductor import config
+config.cpp.enable_kernel_profile = True
+
+######################################################################
+# Following the steps in `Pytorch Profiler <https://pytorch.org/tutorials/recipes/recipes/profiler_recipe.html>`_
+# We are able to get the profiling table and trace files.
+
+from torch.profiler import profile, schedule, ProfilerActivity
+RESULT_DIR = "./prof_trace"
+my_schedule = schedule(
+    skip_first=10,
+    wait=5,
+    warmup=5,
+    active=1,
+    repeat=5)
+
+def trace_handler(p):
+    output = p.key_averages().table(sort_by="self_cpu_time_total", row_limit=20)
+    print(output)
+    p.export_chrome_trace(f"{RESULT_DIR}/{p.step_num}.json")
+
+for _ in range(10):
+    model(**input_dict)  # compiled_model(**input_dict) to get inductor model profiling
+
+total = 0
+with profile(
+    activities=[ProfilerActivity.CPU],
+    schedule=my_schedule,
+    on_trace_ready=trace_handler
+) as p:
+    for _ in range(100):
+        model(**input_dict)  # compiled_model(**input_dict) to get inductor model profiling
+        p.step()
+
+######################################################################
+# We get the following performance profiling table for the eager-mode model:
+#
+# .. image:: ../_static/img/eager_prof.png
+#
+# Similarly, we also get the table for the compiled model with Inductor:
+#
+# .. image:: ../_static/img/inductor_prof.png
+#
+# From the profiling table of the eager model, we can see the most time consumption ops are [aten::addmm, aten::add, aten::copy_, aten::mul, aten::clamp_min, aten::bmm].
+# Comparing with the inductor model profiling table, we notice there are ``mkl::_mkl_linear`` and fused kernel called ``graph_0_cpp_fused_*``. They are the major
+# optimization that the inductor model is doing. Let us discuss them separately.
+#
+# (1) Regard to ``mkl::_mkl_linear``: You may notice the number of calls to this kernel is 362, which is exactly the same as ``aten::linear`` in the eager model profiling table.
+# The CPU total of ``aten::linear`` is 376.888ms, at the mean time it is 231.573ms for ``mkl::_mkl_linear``. This suggests inductor model speed up ~1.63x for the "linear" part.
+# The speed-up majorly comes from we "packed" the ``weight`` tensor to `block memory format <https://oneapi-src.github.io/oneDNN/dev_guide_understanding_memory_formats.html>`_
+# and invoking `cblas_sgemm_compute <https://www.intel.com/content/www/us/en/docs/onemkl/developer-reference-c/2023-1/cblas-gemm-compute-002.html>`_ within Inductor CPU backend
+# to have a better cache beviour during GEMM computation.
+#
+# (2) Regarding non-linear part: The end-to-end latency for the eager/inductor model is 802/339ms. The speed up for the non-linear part is ~3.94x.
```

Comment:
The word "non-linear" usually refers more complicated relationship between values than "linear" while here, you wanted to mean the operations that are not `Linear` op. Suggest you to use a different word like "other ops than linear" or "other memory-intensive ops".

Reply:
Thanks for the advice, done

Reply:
I'm confused. You are still using "non-linear part". Why you mark it "done"?

Reply:
Sorry for that, I forget to merge my branch